src/Umbraco.Web/PublishedCache/ContentCacheAdapter.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Xml.XPath;
+using Umbraco.Core;
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Core.Xml;
+using Umbraco.Web.PublishedCache.NuCache;
+
+namespace Umbraco.Web.PublishedCache
+{
+    internal class ContentCacheAdapter : IPublishedContentCache, IPublishedContentCache2
+    {
+        private readonly IPublishedCache2 _contentCache;
+        private readonly ContentCache _contentCache1;
+        private readonly IContentRouter _contentRouter;
+        private readonly IDomainCache _domainCache;
+
+        public ContentCacheAdapter(ContentCache contentCache, IContentRouter contentRouter,IDomainCache domainCache)
```

Comment:
Should this be `ContentCache` or `IPublishedCache2` ?

Reply:
Introduced IPublishedCache3